yt/data_objects/data_containers.py

```diff
@@ -813,6 +813,19 @@ def create_firefly_object(
             datadir=datadir, clean_datadir=True, **kwargs
         )
 
+        ## Ensure at least one field type contains every field requested
+        kys = self.ds.derived_field_list
+        all_fields = list(zip(*kys))[1]
+        for field in fields_to_include:
+            field = field if not isinstance(field, tuple) else field[1]
+            if field not in all_fields:
+                raise YTFieldNotFound(field=field, ds=self.ds)
+        ## Also generate equivalent of particle_fields_by_type including
+        ## derived fields
+        kysd = defaultdict(list)
+        for k, v in kys:
```

Comment:
in order to apply my previous suggestion
```suggestion
        for k, v in self.ds.derived_field_list:
```